---
title: "Quick start to COSG"
author: "Min Dai"
date: "2021/6/18"
output: rmarkdown::html_vignette
vignette: >
    %\VignetteIndexEntry{Quick start to COSG}
    %\VignetteEngine{knitr::rmarkdown}
    %\VignetteEncoding{UTF-8} 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Overview

COSG is a cosine similarity-based method for more accurate and scalable marker gene identification.

* COSG is a general method for cell marker gene identification across different data modalities, e.g., scRNA-seq, scATAC-seq and spatially resolved transcriptome data.
* Marker genes or genomic regions identified by COSG are more indicative and with greater cell-type specificity.
* COSG is ultrafast for large-scale datasets, and is capable of identifying marker genes for one million cells in less than two minutes.

The method and benchmarking results are described in [Dai et al., (2021)](https://www.biorxiv.org/content/10.1101/2021.06.15.448484v1).


### Installation


##### Install from [GitHub](https://github.com/genecell/COSGR):
```
# install.packages('remotes')
remotes::install_github(repo = 'genecell/COSGR')
```


Load the library:
```{r}
library(COSG) 
library(Seurat) 
```

### Run COSG

```{r}
marker_cosg<-cosg(
  pbmc_small,
  groups='all',
  assay='RNA',
  slot='data',
  mu=1,
  n_genes_user=2000)
```


Check markers:
```{r}
head(marker_cosg$names)
```

Check scores:
```{r}
head(marker_cosg$scores)
```

```{r}
top_list<-c()
for (group in colnames(marker_cosg$names)){
    top_i<-marker_cosg$names[group][1:10,1]
    top_list<-c(top_list,top_i)
}
```

Expression pattern:
```{r fig.height=5, fig.width=16}
DotPlot(pbmc_small, 
        assay = 'RNA',
        features =  unique(top_list)) + RotatedAxis()
```

